### PR TITLE
Close dropdowns when clicking on a file input within

### LINKED ISSUE #16719: clicking a file input no longer closes its parent dropdown in v3.3.5
In bootstrap v3.3.4 if I have a file input inside a `dropdown-menu`, when I clicked the file-input, it would close the `dropdown-menu`.

However in v3.3.5, the click on file-input doesn't close the `dropdown-menu`. 

P.S. I'm assuming bootstrap follows semver in version, because I had `"bootstrap": "~3.3.4"` in my `bower.json` and the behind the scenes patch upgrade was causing me this issue.

### PR DESCRIPTION
**Pull request by archit**
Fixes #16719

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: b01dbd8105bac6eee37e786675dff365327f7ddc
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/68737701

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
Currently waiting on a justification to be given in #16719.

File: js/tests/unit/dropdown.js [hunk 1]
```diff
@@ -388,4 +388,23 @@ $(function () {
 
     assert.ok($dropdown.parent('.btn-group').hasClass('open'), 'dropdown menu is open')
   })
+
+  QUnit.test('should close the dropdown if the user clicks on a file-input', function (assert) {
+    assert.expect(1)
+    var dropdownHTML = '<div class="btn-group">'
+        + '<button type="button" data-toggle="dropdown">Dropdown</button>'
+        + '<ul class="dropdown-menu">'
+        + '<li><input type=\"file\" id="fileInput"></textarea></li>'
```
Comment (cvrebert) [inline js/tests/unit/dropdown.js]: Remove unnecessary backslashes

**Comment by archit** [comment]
@cvrebert Ready for review.

File: js/tests/unit/dropdown.js [hunk 2]
```diff
@@ -388,4 +388,23 @@ $(function () {
 
     assert.ok($dropdown.parent('.btn-group').hasClass('open'), 'dropdown menu is open')
   })
+
+  QUnit.test('should close the dropdown if the user clicks on a file-input', function (assert) {
+    assert.expect(1)
+    var dropdownHTML = '<div class="btn-group">'
+        + '<button type="button" data-toggle="dropdown">Dropdown</button>'
+        + '<ul class="dropdown-menu">'
+        + '<li><input type="file" id="fileInput"></textarea></li>'
```
Comment (cvrebert) [inline js/tests/unit/dropdown.js]: @archit Invalid `</textarea>` here

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 6305552bde1596752a43a0528ae6e2d7aee90950
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/68865091

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0d45f76f56d5050fe63c15bc4751305252afdfcc
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/68867967

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0d45f76f56d5050fe63c15bc4751305252afdfcc
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/68867967

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: js/tests/unit/dropdown.js [hunk 3]
```diff
@@ -388,4 +388,23 @@ $(function () {
 
     assert.ok($dropdown.parent('.btn-group').hasClass('open'), 'dropdown menu is open')
   })
+
+  QUnit.test('should close the dropdown if the user clicks on a file-input', function (assert) {
+    assert.expect(1)
+    var dropdownHTML = '<div class="btn-group">'
+        + '<button type="button" data-toggle="dropdown">Dropdown</button>'
+        + '<ul class="dropdown-menu">'
+        + '<li><input type="file" id="fileInput"></input></li>'
```
Comment (cvrebert) [inline js/tests/unit/dropdown.js]: `</input>` is invalid, just delete it.
`<input>` is an [empty element](https://developer.mozilla.org/en-US/docs/Glossary/empty_element).

**Comment by cvrebert** [comment]
Note to fellow Core Team members: This will need to be squashed if we choose to merge it.

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 18c66ae262109a220e60cd6f44c25511a14a1afb
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/68875363

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
@archit The test is failing on all versions of Internet Explorer. Can you investigate?

**Comment by mdo** [comment]
Closing per my comment in #16719.

**Pull request closed without merging**